### PR TITLE
DRPLDCX-99 Collections

### PR DESCRIPTION
**Pull request by goechsler**
This integrate DC-X collections on the entity browser for images.

It only provides "own" collections, as the DC-X API only supports these atm.

The process is rather complicated as there is a lot of communication with DC-X involved:
1. Retrieve a list of collections and display them as details (say fieldsets)
2. When a detail is opened the list of documents in the respective collection is retrieved
3. The documents need to retrieve their respective preview images (used as draggables for DC-X import).

STR:
- Create a few collections with images, maybe create some sub collections as well.
- Open the entity image browser on an article
- find the first level of your collections hierarchie displayed in the collections fieldset
- click on one of the collection details and watch the images being loaded
- probably sub collections which can be opened as well

### CODE REVIEW

File: src/JsonClient.php [hunk 1]
```diff
@@ -784,4 +784,109 @@ protected function watchdog_exception($type, \Exception $exception, $message = N
...
+  }
+
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getPreview($id) {
+    $json = NULL;
+
+    if ($params == NULL) {
```
Comment (chrfritsch) [inline src/JsonClient.php]: $params is always undefined

File: src/JsonClient.php [hunk 2]
```diff
@@ -784,4 +784,109 @@ protected function watchdog_exception($type, \Exception $exception, $message = N
...
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getCollections() {
+    $params = [
+      'q' => [
+        '_mode' => 'my_usertags',
+        'type_id' => 'usertagtype-default',
+        'parent_id#NULL' => $parent_id,
```
Comment (chrfritsch) [inline src/JsonClient.php]: $parent_id is always undefined

**Comment by chrfritsch** [comment]
I have created a 8.x-2.x into module-dcx-integration for D8.2 compatibility. I have also updated the staging env. So please use the dcx branch of https://github.com/BurdaMagazinOrg/project-dcx-integration

File: src/JsonClient.php (hunk 2 above)
Comment (goechsler) [reply src/JsonClient.php]: Refering to the docs Thorsten Behn sent me $parent_id is supposed to be an empty string. Just replace it with "".

File: src/JsonClient.php (hunk 1 above)
Comment (goechsler) [reply src/JsonClient.php]: True. It's safe to remove the if clause.

**Pull request merged**